safe-global/safe-wallet-web
# refactor: Split up MyAccounts component

## What it solves

Part of [SW-647](https://www.notion.so/safe-global/Refactor-Sidebar-components-1548180fe57380d08e77d05a3a121aec)

## How this PR fixes it

- Renames `AccountsList` to `MyAccounts`
- Creates a new component called `AccountsList`
- Creates `AllSafes`, `PinnedSafes`, `FilteredSafes` components and moves logic
- Creates `AccountsFilter` and `AccountsHeader` components and moves logic
- Renames type `AllSafesGrouped` to `AllSafeItemsGrouped`
- Creates new type callde `AllSafeItems`
- Removes the sorting from filtered safe items
- Changes `QueueActions` to be a button instead of an a element to resolve react error

## ToDos

- [x] Revisit watchlist and pinned safes logic inside `useTrackedSafesCount`
- [x] Write tests for each component

## How to test it

1. Open the accounts page
2. Connect a wallet
3. Observe Safes are listed as before
4. Pin and unpin should still work
5. Pinned safes should be tracked
6. Sidebar should still work as before

## Checklist
